### PR TITLE
refactor: request header generation

### PR DESCRIPTION
**Pull request by NagariaHussain**
* token based auth should work in NodeJS now
* DRYed up code

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Accept: 'application/json',
+    'Content-Type': 'application/json; charset=utf-8',
+  };
+
+  if (useToken && tokenType && token) {
+    headers.Authorization = `${tokenType} ${token()}`;
+  }
+
+  // in case of browser environments
+  if (typeof window != 'undefined') {
```
Comment (nikkothari22) [inline src/utils/index.ts:18]:
```suggestion
  if (typeof window !== 'undefined') {
```

**Review by nikkothari22 — APPROVED** [review]

File: src/auth/index.ts [hunk 2]
```diff
@@ -22,19 +23,15 @@ export class FrappeAuth {
...
-      'X-Frappe-Site-Name': window.location.hostname,
-    };
-
-    if ((window as any).csrf_token && (window as any).csrf_token !== '{{ csrf_token }}') {
-      headers['X-Frappe-CSRF-Token'] = (window as any).csrf_token;
-    }
+    const headers = this.getPreparedRequestHeaders();
 
     return axios
       .post(
```
[the remark this answers is not in the document]
Comment (nikkothari22) [reply src/auth/index.ts:37]: That's a great suggestion! 

I'll check if that won't cause any problems if Axios is used elsewhere in the application (outside of this library), i.e. it doesn't apply globally. I don't think so - there must be a way to create an Axios object and then reuse that everywhere. I'll look into it. 

Thanks!

[the remark this answers is not in the document]
Comment (Alchez) [reply src/auth/index.ts:37]: @nikkothari22, created a PR at #24 that commonifies the request handling

[the remark this answers is not in the document]
Comment (NagariaHussain) [reply src/auth/index.ts:37]: Awesome! Even more DRY!

**Pull request merged**